sw/host/opentitanlib/src/transport/hyperdebug/dfu.rs

```diff
@@ -0,0 +1,433 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// Firmware update protocol for HyperDebug
+
+use anyhow::{bail, Context, Result};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_annotate::Annotate;
+use std::any::Any;
+use std::cell::RefCell;
+
+use crate::transport::{
+    Capabilities, Capability, Progress, Transport, TransportError, UpdateFirmware,
+};
+use crate::util::usb::UsbBackend;
+
+const VID_ST_MICROELECTRONICS: u16 = 0x0483;
+const PID_DFU_BOOTLOADER: u16 = 0xdf11;
+
+const USB_DFU_DETACH: u8 = 0;
+const USB_DFU_DNLOAD: u8 = 1;
+const USB_DFU_GETSTATUS: u8 = 3;
```

Comment:
Similarly, think about whether these constants should be scoped to `HyperdebugDfu`

Reply:
All of these constants are used by public functions in this file, which are not member of any struct.  These functions are used either by the special `HyperdebugDfu` transport, to be used only if strapping with a physical wire has been used to enter the DFU bootloader before opentitantool is invoked, and also used from  the existing `super::Hyperdebug` in the ordinary case of HyperDebug running firmware that can be told to drop into the bootloader.

I do not think it would make sense to try to include these contants in a struct.